src/components/QuizDisplay.tsx

```diff
@@ -1,112 +1,76 @@
-import { useRef, useState } from "react";
+import { useState } from "react";
 import SingleQuestion from "./SingleQuestion";
-import { IAnswerInfo, IQuestionInfo, QuizStatusType } from "../types";
-import Report from "./Report";
-
-interface QuizDisplayProps {
-  allQuestions: IQuestionInfo[];
-  quizStatus: QuizStatusType;
-  setQuizStatus: React.Dispatch<React.SetStateAction<QuizStatusType>>;
-}
+import { IQuizDisplayProps, IAnswerInfo } from "../types";
+import ReturnToQuizSetupBtn from "./ReturnToQuizSetupBtn";
 
 const QuizDisplay = ({
   allQuestions,
-  quizStatus,
   setQuizStatus,
-}: QuizDisplayProps) => {
+  noOfCorrectAnswers,
+  handleReturnToQuizSetup,
+}: IQuizDisplayProps) => {
   const [selectedAnswer, setSelectedAnswer] = useState<number | null>(null);
   const [allAnswers, setAllAnswers] = useState<IAnswerInfo[]>([]);
   const [answerIsSubmitted, setAnswerIsSubmitted] = useState<boolean>(false);
-  const [nextButtonIsDisabled, setNextButtonIsDisabled] =
-    useState<boolean>(true);
+  const [isNextBtnDisabled, setIsNextBtnDisabled] = useState<boolean>(true);
   const [currentQNo, setCurrentQNo] = useState<number>(0);
 
-  const correctAnswers = useRef(0);
-
   const handleSubmitAnswer = () => {
     setAnswerIsSubmitted(true);
-    setNextButtonIsDisabled(false);
-    if (allAnswers[selectedAnswer!].text === correctAnswer)
-      correctAnswers.current++;
+    setIsNextBtnDisabled(false);
+    if (allAnswers[selectedAnswer!].text === correctAnswer.text)
+      noOfCorrectAnswers.current++;
```

Comment:
hmm, not a fan of this, can you explain why we need to drill down a ref, instead of a simple state or even a callback?

Reply:
I had some trouble with using state at some point with re-renders happening that I didn't want and I came across the useRef hook when I was trying to solve this issue. Because I learnt about the useRef hook, knowing that it can be used to hold info without causing re-renders when it changes, I thnk I decided to use this hook. I'll try to refactor it to use state instead